packages/engine/lib/error/src/ext.rs

```diff
@@ -0,0 +1,111 @@
+use core::fmt;
+
+use provider::Provider;
+
+#[cfg(feature = "std")]
+use crate::Report;
+use crate::{Result, ResultExt};
+
+#[cfg(feature = "std")]
+impl<T, E> ResultExt<T> for std::result::Result<T, E>
+where
+    E: std::error::Error + Send + Sync + 'static,
+{
+    type Context = ();
+
+    #[track_caller]
+    fn wrap_err<M>(self, message: M) -> Result<T>
+    where
+        M: fmt::Display + fmt::Debug + Send + Sync + 'static,
```

Comment:
There could be a `Message` trait to reduce code duplication, with a generic `impl` like for the `Context` trait.

Reply:
We had the before but this is just an additional trait which does nothing. The same applies to `Context`.

Reply:
I see, it's to keep the public API smaller.